docs/patterns/sidecar_injection.md

```diff
@@ -4,6 +4,54 @@ Services participating in the service mesh communicate via sidecar proxies insta
 ## Automatic Sidecar Injection
 Automatic sidecar injection is currently the only way to inject sidecars into the service mesh. Sidecars can be automatically injected into applicable Kubernetes pods using a mutating webhook admission controller provided by OSM.
 
-Each OSM instance is given a unique ID on installation. This ID is used while labeling namespaces as a way to configure OSM to monitor the namespaces. When a namespace is labeled with `openservicemesh.io/monitored-by=<mesh-name>`, pods deployed in the monitored namespaces are automatically injected with sidecars by the corresponding OSM instance.
+Automatic sidecar injection can be configured per namespace as a part of enrolling a namespace into the mesh, or later using the Kubernetes API. Automatic sidecar injection can be enabled either on a per namespace or per pod basis by annotating the namespace or pod resource with the sidecar injection annotation. Individual pods and namespaces can be explicitly configured to either enable or disable automatic sidecar injection, giving users the flexibility to control sidecar injection on pods and namespaces.
 
-Since sidecars are automatically injected to pods deployed in OSM monitored namespaces, pods that should not be a part of the service mesh but belong to monitored namespaces need to be explicitly annotated to disable automatic sidecar injection. Using the annotation `"openservicemesh.io/sidecar-injection": "disabled"` on the POD will inform OSM to not inject the sidecar on the POD.
\ No newline at end of file
+### Enabling Automatic Sidecar Injection
+
+Automatic Sidecar injection can be enabled in the following ways:
+
+- While enrolling a namespace into the mesh using `osm` cli: `osm namespace add <namespace> --enable-sidecar-injection`
+
+- Using `kubectl` to annotate individual namespaces and pods to enable sidecar injection:
+
```

Comment:
Should we note here that label `openservicemesh.io/monitored-by=<mesh-name>` is also required on a namespace for automatic sidecar injection?

Reply:
I added this is a prerequisite.